src/Modal/Modal.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from "react";
+import "./modal-style.scss";
+
+interface ModalProps {
+    toggle: boolean;
+    id?: string;
+    fullscreen?: boolean;
+    position?: "left" | "right";
+    className?: string;
+    disableBackdropDismiss?: boolean;
+    ariaLabel?: string;
+    ariaDescribedby?: string;
+    dismissModal?: () => void;
+    header?: React.ReactNode;
+    body?: React.ReactNode;
+    footer?: React.ReactNode;
+}
+
+interface ModalState {
+    open: boolean;
+    close: boolean;
```

Comment:
this state looks useless, maybe use one state `toggle`

Reply:
I think you can rely on the `toggle` prop instead of having a state. I suggest converting this component to `React.FunctionComponent` since removing both states would defeat the purpose of using `React.Component`

this process would remove the need for `constructor`, `componentDidMount` and `componentDidUpdate`